scripts/generate_variable_lists.py

```diff
@@ -0,0 +1,406 @@
+# (C) British Crown Copyright 2025, Met Office.
+# Please see LICENSE.md for license details.
+"""
+This script generates the variable lists for each CMIP experiment for each data request version.
+
+This script scans two source files containing CMIP experiments, their associated variables and the variable metadata
+such as priority level and production labels. Each variable is labelled accordingly and commented out as necessary.
+Each variable list is then saved to a plain text file containing the variables for that experiment.
+
+THIS SCRIPT CURRENTLY CONSIDERS GLOBAL VARIABLES ONLY. NON-GLOBAL VARIABLES ARE FILTERED OUT WITHIN THE FUNCTION
+format_variable_names().
+"""
+
+import argparse
+import json
+import os
+from itertools import chain
+from pathlib import Path
+from typing import Union
+
+IGNORED_PRIORITIES = ("med", "low")
+PRIORITY_ORDER = {"# priority=medium": 1, "# priority=low": 2, "# do-not-produce": 3}
+
+
+def set_arg_parser() -> argparse.Namespace:
```

Comment:
Could you add another argument which is the list of experiments to produce variable lists for (to avoid changing all experiments)